src/rdiff_backup/connection.py

```diff
@@ -288,16 +288,31 @@ def _get(self):
         """Read an object from the pipe and return (req_num, value)"""
         header_string = self.inpipe.read(9)
         if not len(header_string) == 9:
-            raise ConnectionReadError("Truncated header string (problem "
-                                      "probably originated remotely)")
+            raise ConnectionReadError(
+                "Truncated header <{hdr}> "
+                "(problem probably originated remotely)".format(
+                    hdr=header_string))
         format_string = header_string[0:1]
         req_num = self._b2i(header_string[1:2])
         length = self._b2i(header_string[2:])
 
+        if format_string not in b"qobfirRQc":
```

Comment:
I need explanation here. why `qobfirRQc` ? Where did you get this from ?

Reply:
Nevermind, the code was collapse. 
The following seams more explicit to me:
```

if format_string == b"q":
    raise ConnectionQuit("Received quit signal")

try:
        if format_string == b"o":
            result = pickle.loads(self._read(length))
        elif format_string == b"b":
            result = self._read(length)
        elif format_string == b"f":
            result = VirtualFile(self, self._b2i(self._read(length)))
        elif format_string == b"i":
            result = iterfile.FileToMiscIter(
                VirtualFile(self, self._b2i(self._read(length))))
        elif format_string == b"r":
            result = self._getrorpath(self._read(length))
        elif format_string == b"R":
            result = self._getrpath(self._read(length))
        elif format_string == b"Q":
            result = self._getqrpath(self._read(length))
        elif format_string == b"c":
            result = Globals.connection_dict[self._b2i(self._read(length))]
        else:
            raise ConnectionReadError(
                "Format character '{form}' invalid in <{hdr}> "
                "(problem probably originated remotely)".format(
                    hdr=header_string, form=format_string))
except MemoryError:
    raise ConnectionReadError(
        "Impossibly high data amount evaluated in header <{hdr}> "
        "(problem probably originated remotely)".format(
            hdr=header_string))
```

So whenever we are adding a new formatString, we don't need to "think" about adding it in the validation.

Reply:
Good point with the maintenance. I actually wanted to have the simplest check first, but it's just cosmetic, I'll change it.

Reply:
OK, addressed.
I preferred to keep the `_read` separate, IMHO it makes a more readable code, and also one easier to debug (I can check the content of data before it goes into the `_getxxx` function).